static/app/components/searchQueryBuilder/tokens/useSortedFilterKeyItems.tsx

```diff
@@ -261,6 +265,32 @@ export function useSortedFilterKeyItems({
         type: 'section',
       };
 
+      const shouldShowMatchKeySuggestions =
+        !disallowFreeText &&
```

Comment:
Why does this need to be disabled when free text is disallowed? Seems like it would still be useful in that scenario